src/solvers/base_solver.cpp

```diff
@@ -55,57 +44,56 @@ void BaseSolver::SetNaturalBCs(mfem::Array<int> &nat_bdr, mfem::Coefficient *nat
   m_nat_bdr_coef = nat_bdr_coef;
 }
 
-void BaseSolver::SetState(const mfem::Array<FiniteElementState> &state)
+void BaseSolver::SetState(const std::vector<FiniteElementState> &state)
 {
   MFEM_ASSERT(state.Size() > 0, "State vector array of size 0 in BaseSolver::SetState.");
   m_state = state;
 
-  MPI_Comm_rank(m_state[0].space->GetComm(), &m_rank);
+  MPI_Comm_rank(m_state.begin()->space->GetComm(), &m_rank);
```

Comment:
Please don't use the iterator unless you are iterating...

Reply:
oops, this was leftover from when I tried out std::map, so we had no m_states[0]! Fixed in upcoming code-style PR.